pages/api/changePassword.ts

```diff
@@ -0,0 +1,43 @@
+import { withApiAuth } from "../../lib/auth/withApiAuth";
+import { withApiMethods } from "../../lib/withApiMethods";
+import { withApiValidation } from "../../lib/withApiValidation";
+import { hash, compare } from "../../lib/utils/bcryptHash";
+import { prisma } from "../../lib/db";
+import { Asserts, object, string } from "yup";
+
+export const changePasswordSchema = object({
+  oldPassword: string().required("Old password is required"),
+  newPassword: string().required("New password is required"),
+  hashedOldPassword: string().required(),
+  id: string().required("User id is required"),
+});
+
+export type ChangePasswordType = Asserts<typeof changePasswordSchema>;
+
+export default withApiAuth(
+  withApiMethods({
+    POST: withApiValidation({ body: changePasswordSchema }, async (req, res) => {
```

Comment:
Maybe we should use the `PATCH` method here?

Reply:
I was talking about it with @AdamSiekierski, and he said we will better do it with POST method :smile:

Reply:
Ah, okay then 👍

Reply:
POST is an appropriate method here. Only PUT may be discussable, but I'd stick to POST in this case (this endpoint isn't RESTful anyways, and it doesn't really make sense to make it so)